core/src/main/scala-2/chisel3/Aggregate.scala

```diff
@@ -0,0 +1,333 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3
+
+import scala.language.experimental.macros
+import chisel3.experimental.SourceInfo
+import chisel3.internal.HasId
+import chisel3.internal.sourceinfo.{SourceInfoTransform, VecTransform}
+
+/** An abstract class for data types that solely consist of (are an aggregate
+  * of) other Data objects.
+  */
+sealed trait Aggregate extends AggregateImpl
```

Comment:
why do we need this trait here?

Reply:
Because it's sealed and [annoyingly] the concrete extensions have to be in the same file.